src/server.c

```diff
@@ -2737,7 +2737,7 @@ void initServer(void) {
     evictionPoolAlloc(); /* Initialize the LRU keys pool. */
     server.pubsub_channels = dictCreate(&keylistDictType);
     server.pubsub_patterns = dictCreate(&keylistDictType);
-    server.pubsubshard_channels = dictCreateMultiple(&keylistDictType, slotCount);
+    server.pubsubshard_channels = zcalloc(sizeof(dict *) * slotCount);
```

Comment:
I was thinking we can postpone even this allocation (128KB) but this is fine. It is not too much.

It's good we can avoid creating the dicts, which is roughly 1M if I calculate correctly (16K * 64).

Reply:
I prefer to just leave it like this, it is more readable since people can understand the data structure from the init function.